packages/grid/_modules_/grid/models/colDef/gridColDef.ts

```diff
@@ -110,12 +111,12 @@ export interface GridColDef {
    * Allows to override the component rendered as cell for this column.
    * @param params
    */
-  renderCell?: (params: GridCellParams) => React.ReactNode;
+  renderCell?: (params: GridRenderCellParams) => React.ReactNode;
   /**
    * Allows to override the component rendered in edit cell mode for this column.
    * @param params
    */
-  renderEditCell?: (params: GridCellParams) => React.ReactNode;
+  renderEditCell?: (params: GridRenderCellParams) => React.ReactNode;
```

Comment:
We can reuse `GridRenderCellParams` but it also need to include `GridEditCellProps` or the user won't be able to access the additional props.

https://github.com/mui-org/material-ui-x/blob/019c7ccb19a8ee31214da7936cdd4c899921aa3c/packages/grid/_modules_/grid/models/gridEditRowModel.ts#L3-L6

```suggestion
  renderEditCell?: (params: GridRenderEditCellParams) => React.ReactNode;
```